app.js

```diff
@@ -1,15 +1,30 @@
+// State
+let volume = 10;
 
-// state
+// Helper functions
+function clamp(value, min, max) {
+    return Math.min(max, Math.max(min, value));
+}
 
+// Counter component
+const counterRoot = document.getElementById('volume-counter');
+const [downButton, upButton] = counterRoot.querySelectorAll('button');
+const volumeDisplay = counterRoot.querySelector('span');
 
-// component
-    // define and grab DOM elements
-    // display functions
-    // subscribe to events
-        // event handlers - what needs to happen?
-        // logic and calculations
-        // state update
-        // re-display components
+function displayCounter() {
+    volumeDisplay.textContent = volume;
+}
 
+function getChangeVolumeCallback(delta) {
+    return () => {
+        volume = clamp(volume + delta, 1, 20);
+        displayCounter();
+    };
+}
```

Comment:
Nice higher order function. We sometimes write these as double arrow functions:

```js
const getCallback = delta => () => {
    volume = clamp(volume + delta, 1, 20);
    displayCounter();
};

Reply:
Oh that's slick.